src/main/java/seedu/address/logic/commands/UploadCommand.java

```diff
@@ -0,0 +1,132 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_PERSONS;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import javax.swing.JFileChooser;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.ProfilePicFilePath;
+
+/**
+ * Uploads a profile picture for a specified person
+ */
+public class UploadCommand extends Command {
+
+    public static final String COMMAND_WORD = "upload";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Uploads a profile picture from the computer"
+            + "by the index number used in the last person listing. "
+            + "Existing profile picture  will be overwritten by the input.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "Example: " + COMMAND_WORD + " 1 ";
+
+    public static final String MESSAGE_UPLOAD_PERSON_SUCCESS = "Uploaded Profile Picture for Person: %1$s";
+    public static final String MESSAGE_UPLOAD_CANCELLED = "Cancelled profile picture upload for Person: %1$s";
+    public static final String MESSAGE_UPLOAD_ERROR = "Error getting profile picture. "
+            + "Did you ensure it is a .png file?";
+    private final Index targetIndex;
+
+    /**
+     * Default constructor for a UploadCommand object
+     * @param targetIndex Index to be deleted
```

Comment:
is this "deleted" because the target will be replaced?

Reply:
Nope, typo on my end. Will change to be "uploaded"